tests/test_coverity.py

```diff
@@ -1,126 +1,207 @@
+
+        # Login to Coverity
+        coverity_conf_service.login("user", "password")
+
+        with open("tests/columns_keys.json", "rb") as content:
+            mock_get.return_value.content = content.read()
+        mock_get.return_value.ok = True
+        coverity_conf_service.retrieve_checkers()
+        mock_get.assert_called_once()
+
+    def test_get_defects_call(self):
+        with open("tests/columns_keys.json", "r") as content:
+            column_keys = json.loads(content.read())
+
+        self.filters = {
+            "checker": None,
+            "impact": None,
+            "kind": None,
+            "classification": None,
+            "action": None,
+            "component": None,
+            "cwe": None,
+            "cid": None,
+        }
+        self.fake_json = {"test": "succes"}
+        self.fake_checkers = {
+            "checkerAttribute": {"name": "checker", "displayName": "Checker"},
+            "checkerAttributedata": [{"key": "checker_key", "value": "checker_value"}],
+        }
+
+        self.fake_stream = "test_stream"
+        # initialize what needed for the REST API
+        coverity_service = self.initialize_coverity_service(login=True)
+
+        with requests_mock.mock() as mocker:
+            mocker.get(self.column_keys_url, json=column_keys)
+            mocker.get(self.checkers_url, json=self.fake_checkers)
+            mocker.get(self.stream_url, json={"stream": "valid"})
+            mocker.post(self.issues_url, json=self.fake_json)
+
+            # Validate stream name
+            coverity_service.validate_stream(self.fake_stream)
+            # Retrieve column keys
+            assert coverity_service.retrieve_column_keys()["Issue Kind"] == "displayIssueKind"
+            assert coverity_service.retrieve_column_keys()["CID"] == "cid"
+            # Retrieve checkers
+            assert coverity_service.retrieve_checkers() == ["checker_key"]
+            # Get defects
+            assert coverity_service.get_defects(self.fake_stream, self.filters, column_names=["CID"]) == self.fake_json
+            # Total amount of request are 4 => column keys, checkers, stream and defects/issues
+            assert mocker.call_count == 4
+
+            # check get requests
+            get_urls = [self.stream_url, self.column_keys_url, self.checkers_url]
+            for index in range(len(get_urls)):
+                mock_req = mocker.request_history[index]
+                assert mock_req.url == get_urls[index]
+                assert mock_req.verify
+                assert mock_req.headers["Authorization"] == requests.auth._basic_auth_str("user", "password")
+            # check post request (last request)
+            assert mocker.last_request.url == self.issues_url
+            assert mocker.last_request.verify
+            assert mocker.last_request.headers["Authorization"] == requests.auth._basic_auth_str("user", "password")
```

Comment:
This test case does not actually test a lot. To really test `get_defects`, you should validate the value of `data` it built: https://github.com/melexis/sphinx-coverity-extension/blob/17750dca53e4dc6cd467759e8c19be5aca0441e8/mlx/coverity_services.py#L313

Reply:
Resolved in 7b78da6f546848e08ed586691e9a64ea61d9a38d